components/Settings/SizeSelect.tsx

```diff
@@ -0,0 +1,52 @@
+import { FC, useState, useContext, useEffect } from "react";
+import ImageExample from "./ImageExample";
+import styles from "../../styles/SizeSelect.module.scss";
+import QRSizeContext from "../../context/Context";
+
+const SizeSelect: FC = () => {
+    const [ demo_size, setDemoSize ] = useState(360);
+    const { example_size, setExampleSize } = useContext(QRSizeContext);
+    const side_multiplier = 45;
+
+    useEffect(() => {
+        setDemoSize(example_size * side_multiplier);
+    }, [example_size])
+
+    const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const size = e.target.value;
+
+        switch (size) {
+            case "small":
+                setDemoSize(180)
+                setExampleSize(4)
+                break;
+            case "medium":
+                setDemoSize(270)
+                setExampleSize(6)
+                break;
+            case "large":
+                setDemoSize(360)
+                setExampleSize(8)
+                break;
+            default:
+                break;
```

Comment:
no default size?

Reply:
I'm wondering if switch is needed, or maybe it should set those sizes in default too, just in case to support some undefined typeerrors

Reply:
Added default case.